test/test_autograd.py

```diff
@@ -4129,6 +4130,877 @@ def fn(*inputs):
+            self.assertTrue(torch.is_tensor(res))
+            self.assertEqual(res.size(), base1.size() + base2.size())
+        elif isinstance(base1, tuple) and torch.is_tensor(base2):
+            self.assertTrue(isinstance(res, tuple))
+            self.assertEqual(len(res), len(base1))
+            for el_res, el_base1 in zip(res, base1):
+                self.assertTrue(torch.is_tensor(el_res))
+                self.assertTrue(torch.is_tensor(el_base1))
+                self.assertEqual(el_res.size(), el_base1.size() + base2.size())
+        elif torch.is_tensor(base1) and isinstance(base2, tuple):
+            self.assertTrue(isinstance(res, tuple))
+            self.assertEqual(len(res), len(base2))
+            for el_res, el_base2 in zip(res, base2):
+                self.assertTrue(torch.is_tensor(el_res))
+                self.assertTrue(torch.is_tensor(el_base2))
+                self.assertEqual(el_res.size(), base1.size() + el_base2.size())
+        elif isinstance(base1, tuple) and isinstance(base2, tuple):
+            self.assertTrue(isinstance(res, tuple))
+            self.assertEqual(len(res), len(base1))
+            for el_res, el_base1 in zip(res, base1):
+                self.assertTrue(isinstance(el_res, tuple))
+                self.assertEqual(len(res), len(base2))
+                for el_el_res, el_base2 in zip(el_res, base2):
+                    self.assertTrue(torch.is_tensor(el_el_res))
+                    self.assertTrue(torch.is_tensor(el_base2))
+                    self.assertEqual(el_el_res.size(), el_base1.size() + el_base2.size())
+        else:
+            # Wrong bases
+            raise RuntimeError("The bases given to `_assert_interleaved_struct` don't have"
+                               " the right structure.")
+
+    def test_vjp_err_check(self):
+        def foo(a):
+            return 3 * a.narrow(0, 0, 3)
+
+        def bar(a):
+            return 3 * a.narrow(0, 0, 3), "bar"
+
+        inp = torch.rand(4)
+        with self.assertRaisesRegex(TypeError, "The inputs given to vjp must be either a Tensor"):
+            res = autogradF.vjp(foo, (inp, 2))
+
+        with self.assertRaisesRegex(TypeError, "The outputs of the user-provided function given to vjp must"):
+            res = autogradF.vjp(bar, inp)
+
+        with self.assertRaisesRegex(RuntimeError, "The vector v can only be None if the user-provided function returns"):
+            res = autogradF.vjp(foo, inp)
+
+        with self.assertRaisesRegex(RuntimeError, "v is a tuple of invalid length: should be 1 but got 2."):
+            res = autogradF.vjp(foo, inp, (torch.ones_like(inp), torch.ones_like(inp)))
+
+        with self.assertRaisesRegex(RuntimeError, "Entry 0 in v has invalid size: should be torch.Size"):
+            res = autogradF.vjp(foo, inp, torch.ones(2))
+
+        res = autogradF.vjp(foo, inp, torch.ones(3))[1]
+        self._assert_same_struct(res, inp)
+
+    def test_vjp_err_check_strict(self):
```

Comment:
I don't see a test for "The jacobian of the user-provided function is independent of..." (the case where we are strict, have create_graph=True, but the vjp doesn't require grad)

Reply:
Good catch, this error was added at the end and I forgot to update the tests. Will do